File: android/app/src/main/kotlin/com/transport_system/ts_driver/agora/enums/CallState.kt
package com.transport_system.ts_driver.agora.enums

enum class CallState(private val displayName: String) {
    IDLE("Calling..."),
    CALLING("Calling..."),
    RINGING("Ringing..."),
    DECLINED("Declined"),
    USER_BUSY("User Busy"),
    CONNECTING("Connecting..."),
    CONNECTED("Connected"),
    FAILED("Failed"),
    NOT_ANSWERED("Not Answered");

    // Method to get the display name
    fun getName(): String {
        return displayName
    }
}

File: ios/Runner/Agora/Enums/CallState.swift
//
//  CallState.swift
//  Runner
//
//  Created by Hashim Khan on 24/01/2024.
//

enum CallState {
    case idle
    case calling
    case ringing
    case declined
    case userBusy
    case connecting
    case connected
    case failed
    case notAnswered
    
    
    // Method to get the name of the enum case as a string
    func getName() -> String {
        switch self {
        case .idle:
            return "Calling..."
        case .calling:
            return "Calling..."
        case .ringing:
            return "Ringing..."
        case .declined:
            return "Declined"
        case .userBusy:
            return "User Busy"
        case .connecting:
            return "Connecting..."
        case .connected:
            return "Connected"
        case .failed:
            return "Failed"
        case .notAnswered:
            return "Not Answered"
        }
    }
}
